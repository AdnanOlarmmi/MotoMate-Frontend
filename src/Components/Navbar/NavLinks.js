import React from 'react';

const NavLinks = () => {
  return (
    <ul className="mt-4 text-lg sm:text-2xl">
      <li className="pl-4 py-2 hover:bg-gray-100">
        <a href="#">Home</a>
      </li>
      <li className="pl-4 py-2 hover:bg-gray-100">
        <a href="#">Reservations</a>
      </li>
      <li className="pl-4 py-2 hover:bg-gray-100">
        <a href="#">Add a Motorcycle</a>
      </li>
      <li className="pl-4 py-2 hover:bg-gray-100">
        <a href="#">Contact</a>
      </li>
    </ul>
  );
};

export default NavLinks;
