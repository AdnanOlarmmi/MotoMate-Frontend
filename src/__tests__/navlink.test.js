import renderer from 'react-test-renderer'

import { BrowserRouter } from 'react-router-dom'

import NavLinks from '../Components/Navbar/NavLinks'

import userEvent from '@testing-library/user-event';
// eslint-disable-next-line
import { render, screen } from '@testing-library/react';

describe("Tests Navbar component",() => {
    it("Should render Navbar Component",()=>{
        const tree = renderer.create(<BrowserRouter>
            <NavLinks />
        </BrowserRouter>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    const list = screen.getByRole('list',{
        hidden:true
    }).childElementCount;
    expect(list).toEqual(5);
    const link = screen.getByText('Motorcycle');
    userEvent.click(link);
    expect(link.closest('a')).toHaveAttribute('class', 'active');
})
