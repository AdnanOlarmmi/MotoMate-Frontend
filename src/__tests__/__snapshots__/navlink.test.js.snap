// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Navbar component Should render Navbar Component 1`] = `
<ul
  className="mt-4 w-full text-lg sm:text-xl"
>
  <li
    className="pl-4 py-2 hover:bg-cyan-900 hover:text-white"
  >
    <a
      className=""
      href="/motorcycles"
      onClick={[Function]}
    >
      Motorcycles
    </a>
  </li>
  <li
    className="pl-4 py-2 hover:bg-cyan-900 hover:text-white"
  >
    <a
      className=""
      href="/reservations"
      onClick={[Function]}
    >
      My Reservations
    </a>
  </li>
  <li
    className="pl-4 py-2 hover:bg-cyan-900 hover:text-white"
  >
    <a
      className=""
      href="/add-motorcycle"
      onClick={[Function]}
    >
      Add a Motorcycle
    </a>
  </li>
  <li
    className="pl-4 py-2 hover:bg-cyan-900 hover:text-white"
  >
    <a
      className=""
      href="/delete-motorcycle"
      onClick={[Function]}
    >
      Delete Motorcycle
    </a>
  </li>
  <li
    className="pl-4 py-2 hover:bg-cyan-900 hover:text-white"
  >
    <a
      className=""
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </li>
</ul>
`;
